test/demo/findIntent.inttest.ts

```diff
@@ -0,0 +1,142 @@
+import 'jest';
+import fetch from 'node-fetch';
+
+import {Application as DirectoryApp} from '../../src/client/directory';
+import {Context, AppIntent} from '../../src/client/main';
+
+import * as fdc3Remote from './utils/fdc3RemoteExecution';
+import {delay} from './utils/delay';
+import {setupStartNonDirectoryAppBookends, setupOpenDirectoryAppBookends, setupTeardown} from './utils/common';
+import {testManagerIdentity, testAppInDirectory1, testAppNotInDirectory1} from './constants';
+
+const validIntent = 'DialCall';
+
+const invalidIntent = 'some-nonexistent-intent';
+
+setupTeardown();
+
+describe('Resolving applications by intent', () => {
+    let directory: DirectoryApp[];
+
+    beforeAll(async () => {
+        // TODO: Replace this hard-coded string once possible (SERVICE-392?)
+        // Fecth and store the current app directory
+        directory = await fetch('http://localhost:3923/test/sample-app-directory.json').then(response => response.json());
+    });
+
+    describe('Without ad-hoc apps running', () => {
+        test('When calling resolve with an intent which at least one directory application accepts, \
+the returned object lists all and only those apps which accept the intent', async () => {
+            // Query the actual app directory to find all apps which should be returned
+            const expectedAppNames = getDirectoryAppsForIntent(directory, validIntent);
+
+            // Resolve the valid intent
+            const appIntent = await findIntent(validIntent);
+
+            // Compare the two arrays (sorted as order doesn't matter)
+            expect(extractSortedAppNames(appIntent)).toEqual(expectedAppNames);
+        });
+
+        test('When resolving an intent without a specific context, the returned object lists all \
+directory applications accepts the intent regardless of their context specifications.', async () => {
```

Comment:
"directory applications accepts" -> "directory applications that accept"

Reply:
👍